doc/Type/List.pod6

```diff
@@ -884,24 +884,40 @@ Defined as:
     multi method reduce(Any:D: &with)
     multi sub reduce (&with, +list)
 
-The first form is obviously a no-op. The second form generates a single
+The first form is obviously a no-op. The remaining forms generate a single
 "combined" value from a list of arbitrarily many values, by iteratively
-applying a function which knows how to combine I<two> values.
+applying a function which knows how to combine I<two> values. In addition to
+the subroutine and the list, an initial value can be provided to initialize
+the reduction, which ends up being the return value if the list is empty.
+Thus C<reduce f, init, list> combines the elements of the list from left
+to right, as follows:
 
-If C<@values> contains just a single element, the operator is applied to that
+    result0 = init
+    result1 = f(result0, list[0])
+    result2 = f(result1, list[1])
+    ...
+    resultn = f(resultn-1, list[n-1])
+
+C<resultn> is the final result for an n-element list.
+
+    say reduce &[+], [1, 2, 3];     # OUTPUT: «6␤»
+    say [1, 2, 3].reduce: &[+];     # OUTPUT: «6␤»
+    say reduce &max, [5, 9, 12, 1]; # OUTPUT: «12␤»
+
+If C<list> contains just a single element, the operator is applied to that
 single element if possible; if not, it returns the element itself.
 
-    say [-] <10 5 3>; #OUTPUT: 2␤
-    say [-] 10;       #OUTPUT: 10␤
+    say reduce &[+], <10 5 3>; # OUTPUT: «2␤»
```

Comment:
The initial example was bad, because it was using the hyperoperator instead of the reduce routine, but it made sense by itself. Again, this one is not really a good example.